mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -2759,12 +2759,16 @@ pub fn max_taker_vol_from_available(
     rel: &str,
     min_tx_amount: &MmNumber,
 ) -> Result<MmNumber, MmError<MaxTakerVolumeLessThanDust>> {
-    let dex_fee_rate = dex_fee_rate(base, rel);
-    let threshold_coef = &(&MmNumber::from(1) + &dex_fee_rate) / &dex_fee_rate;
-    let max_vol = if available > min_tx_amount * &threshold_coef {
-        available / (MmNumber::from(1) + dex_fee_rate)
-    } else {
+    let max_vol = if base == "KMD" || rel == "KMD" {
         &available - min_tx_amount
+    } else {
+        let dex_fee_rate = dex_fee_rate(base, rel);
+        let threshold_coef = &(&MmNumber::from(1) + &dex_fee_rate) / &dex_fee_rate;
+        if available > min_tx_amount * &threshold_coef {
+            available / (MmNumber::from(1) + dex_fee_rate)
+        } else {
+            &available - min_tx_amount
+        }
```

Comment:
I suppose this should be reverted, as KMD zero dexfee is only for TPU?

Reply:
yeah done. Thanks